pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+                raise ValueError('Schema url %s not found. Exception: %s' %
+                                 (self._mof_zip_url, ie))
+
+            with open(self._mof_zip_filename, 'wb') as fp:
+                for data in ufo:
+                    fp.write(data)
+
+        if not os.path.isdir(self.schema_mof_dir):
+            if_first("Creating directory for CIM Schema MOF files: %s" %
+                     self.schema_mof_dir)
+            os.mkdir(self.schema_mof_dir)
+        if not os.path.isfile(self._schema_mof_filename):
+            if_first("Unpacking CIM Schema archive: %s" %
+                     self._mof_zip_filename)
+
+            zfp = None
+            try:
+                zfp = ZipFile(self._mof_zip_filename, 'r')
+                nlist = zfp.namelist()
+                for file_ in nlist:
+                    dfile = os.path.join(self.schema_mof_dir, file_)
+                    if dfile[-1] == '/':
+                        if not os.path.exists(dfile):
+                            os.mkdir(dfile)
+                    else:
+                        with open(dfile, 'w+b') as dfp:
+                            dfp.write(zfp.read(file_))
+            finally:
+                if zfp:
+                    zfp.close()
+
+    def build_schema_mof(self, schema_classes):
+        """
+        Build a mof file that includes the pragmas for the DMTF schema CIM
+        classes defined in 'schema_classes' using the DMTF CIM schema defined by
+        the directory 'schema_dir' and the schema version defined by
+        'schema_ver'.
+
+        It builds a compilable mof string in the form:
+
+            pragma locale ("en_US")
+            pragma include ("System/CIM_ComputerSystem.mof")
+
+        Parameters:
+
+          schema_classes (:term:`py:list` of :term:`string`):
+            These must be classes in the define DMTF schema and can be just
+            a list of the leaf classes required for a working class repository.
+            If the returned string is compiled, the mof compiler will search
+            the directory defined by schema_dir for dependent classes.
+
+        Returns:
+            :term:`string`. valid MOF containing pragma statements defining
+            all of the classes in 'schema_classes' that can be compiled with
+            the MOF compiler.be used to compile the classes
+
+        Raises:
+            ValueError if classname in 'schema_classes' not in the DMTF
+            schema installed
+        """
+        if isinstance(schema_classes, six.string_types):
+            schema_classes = [schema_classes]
+
+        schema_lines = []
+        with open(self.schema_mof_filename, 'r') as f:
+            schema_lines = f.readlines()
+        f.close()
+
+        output_lines = ['#pragma locale ("en_US")\n']
+        for cln in schema_classes:
+            test_cln = '/%s.mof' % cln
+            found = False
+            for line in schema_lines:
+                if line.find(test_cln) != -1:
```

Comment:
This algorithm works only if the class names are unique within the schema. Since this class is used only for the DMTF schema, that is the case, but maybe it is worthwhile stating in a comment that we depend on that.

Reply:
We noted in the overview that this whole class only applies to DMTF schema. Since we are installing this into a single namespace, that is a given anyway.

Reply:
DONE